File: interface/Transaction.ts
export interface IPayee {
  payeeId?: number;
  userAccountNumber: string;
  payeeName: string;
  payeeAccountNumber: string;
  bankName: string;
}

export interface Transaction {
  transactionId?: number;
  fromAccount: string;
  toAccount: string;
  amount: number;
  remarks: string;
  transactionType?: string;
  status?: string;
  createdAt?: string;
}

export interface TransferRequest {
  fromAccount: string;
  toAccount: string;
  amount: number;
  remarks: string;
}

File: transactionms/src/main/java/com/wipro/transactionms/entity/Transaction.java
package com.wipro.transactionms.entity;

import java.time.LocalDateTime;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "transactions")
@Data
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int transactionId;

    private String fromAccount;

    private String toAccount;

    private double amount;

    private String remarks;

    private String transactionType; 

    private String status; 

    private LocalDateTime createdAt;
}